vcd/sample_vcd_test_config.json

```diff
@@ -90,6 +90,44 @@
     "uploadPieceSize": 5,
     "uploadProgress": true
   },
+  "testEnvBuild": {
+    "//": "This section is optional for running tests. It's only needed to build the test environment in a new vCD",
+
+    "// ": "Entities that will be created in an empty vCD use names defined above",
+
+    "// ": "vcd.org                    -> organization name",
+    "// ": "vcd.vdc                    -> VDC name",
+    "// ": "vcd.catalog.name           -> catalog name",
+    "// ": "vcd.catalog.catalogItem    -> catalog item name",
+    "// ": "networking.edgeGateway     -> edge gateway name",
+    "// ": "networking.externalNetwork -> external network name",
+
+    "//": "Values used to build a new environment. Build will fail if the next 5 ones are empty",
+    "gateway": "10.15.21.1",
+    "netmask": "255.255.224.0",
+    "externalNetworkStartIp": "10.15.21.10",
+    "externalNetworkEndIp": "10.15.21.11",
+    "dns1": "10.15.21.101",
+    "dns2": "8.8.4.4",
+
+    "//": "external network may be different from the one defined for testing",
+    "externalNetworkPortGroup": "network_name_from_vcenter",
+    "externalNetworkPortGroupType": "NETWORK",
```

Comment:
The comment above says "external network may be different", but it's unclear whether these settings can actually be the same. Imagine that my goal is just to create an env, run tests in it and then leave the env for my manual testing. For this common use case one external network should be enough. Maybe clarify the comment?

Reply:
OK, this workflow really needs some more cleanup, I just got a related error about storage profile and am puzzled why can't they be the same in JSON or why should I define the second one as a different one (I only see the result in the template that this caused duplication):

```
  # vcd_org_vdc.myvdc will be created
  + resource "vcd_org_vdc" "myvdc" {
      + allocation_model         = "AllocationVApp"
      + allow_over_commit        = true
      + delete_force             = true
      + delete_recursive         = true
      + enable_fast_provisioning = true
      + enable_thin_provisioning = true
      + enabled                  = true
      + id                       = (known after apply)
      + name                     = "myvdc"
      + network_pool_name        = "vc1-TestbedCluster-09:13:18-VXLAN-NP"
      + network_quota            = 50
      + org                      = "myorg"
      + provider_vdc_name        = "vc1-TestbedCluster-09:13:18"

      + compute_capacity {
          + cpu {
              + limit = 0
            }

          + memory {
              + limit = 0
            }
        }

      + storage_profile {
          + default = true
          + enabled = true
          + limit   = 0
          + name    = "Development"
        }
      + storage_profile {
          + default = false
          + enabled = true
          + limit   = 0
          + name    = "Development"
        }
    }

Plan: 10 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.

# terraform apply -auto-approve
vcd_org.myorg: Creating...
vcd_external_network.my-ext-network: Creating...
vcd_org.myorg: Creation complete after 2s [id=7e2bc507-aa44-4623-bc11-68d935651932]
vcd_org_vdc.myvdc: Creating...
vcd_external_network.my-ext-network: Still creating... [10s elapsed]
vcd_external_network.my-ext-network: Creation complete after 15s [id=my-ext-network]

Error: error creating vdc: &errors.errorString{s:"error retrieving vdc: API Error: 400: [ 0e50d451-a760-4339-9f59-b69b5d4af18e ] The request has duplicate VDC storage policy \"Development\"."}
```

Reply:
> The comment above says "external network may be different", but it's unclear whether these settings can actually be the same. Imagine that my goal is just to create an env, run tests in it and then leave the env for my manual testing. For this common use case one external network should be enough. Maybe clarify the comment?

Changed the comment, saying that they MUST be different

Reply:
> OK, this workflow really needs some more cleanup, I just got a related error about storage profile and am puzzled why can't they be the same in JSON or why should I define the second one as a different one (I only see the result in the template that this caused duplication):

The name of the field in `testEnvBuild` is `StorageProfile2` because we have already a storage profile field for terraform tests, which only need one. We need to add the second one to VDC, though, to be used in govcd tests. I added more comments to make this more clear.